app/script/view_model/ConversationInputEmojiViewModel.js

```diff
@@ -286,56 +290,68 @@ z.ViewModel.ConversationInputEmojiViewModel = class ConversationInputEmojiViewMo
   }
 
   _update_emoji_popup(input) {
-    if (!input.value) {
+    const text = input.value;
+    if (!text) {
       return;
     }
 
-    const query = input.value.substr(this.emoji_start_pos, input.selectionStart - this.emoji_start_pos);
-    if (query.length < QUERY_MIN_LENGTH || query[0] === ' ' || /\s{2,}/.test(query) || this.emoji_list.length === 0) {
-      this.emoji_div.remove();
-    } else {
-      const query_words = query.split(' ');
-      const emoji_matched = this.emoji_list
-        .filter((emoji) => {
-          const emoji_name_words = emoji.name.split(' ');
-          return query_words.every((query_word) => emoji_name_words.some((emoji_name_word) => emoji_name_word.startsWith(query_word)));
-        })
-        .reduce((acc, emoji, index) => {
-          if (!acc.find((item) => item.icon === emoji.icon)) {
-            acc.push(emoji);
-          }
-          return acc;
-        }, [])
-        .sort((emoji_a, emoji_b) => {
-          const usage_count_a = this._get_usage_count(emoji_a.name);
-          const usage_count_b = this._get_usage_count(emoji_b.name);
-          if (usage_count_a === usage_count_b) {
-            return z.util.StringUtil.sort_by_priority(emoji_a.name, emoji_b.name, query);
-          }
-          return usage_count_b - usage_count_a;
-        })
-        .slice(0, EMOJI_LIST_LENGTH)
-        .map((emoji) => `<div class='emoji'><span class='symbol'>${emoji.icon}</span><span class='name'>${emoji.name}</span></div>`)
-        .join('');
-
-      if (emoji_matched === '') {
-        this._close_emoji_popup();
-      } else {
-        window.addEventListener('click', this.remove_emoji_popup);
-        this.emoji_div
-          .html(emoji_matched)
-          .appendTo('body')
-          .show();
-        this.emoji_div.find('.emoji:nth(0)').addClass('selected');
-
-        const pos = this._get_cursor_pixel_pos(input);
-        const top = pos.top - this.emoji_div.height() - EMOJI_LIST_OFFSET_TOP;
-        const left = pos.left - EMOJI_LIST_OFFSET_LEFT;
-
-        this.emoji_div.css('left', left);
-        this.emoji_div.css('top', top);
-      }
+    const query = text.substr(this.emoji_start_pos, input.selectionStart - this.emoji_start_pos);
```

Comment:
Since you need `input.selectionStart` here anyway, you could as well use this fancy way of writing on line 293 🙂
```
const {selectionStart: selection, value: text} = input;
```

Reply:
Thanks for the hint. Added the fancy way in order to shorten some of the lines and remove redundancy in the writing. Tried to address another bug here but gave up on it eventually for now...